<think>
No reasoning found</think>
```java
        public void uncaughtException(Thread thread, Throwable throwable) {
            jobControlExceptionStackTrace = Utils.getStackStraceStr(throwable);
            try {
                jobControlException = getExceptionFromString(jobControlExceptionStackTrace);
            } catch (Exception e) {
                String errMsg = "Could not resolve error that occured when launching job: "
                        + jobControlExceptionStackTrace;
                jobControlException = new RuntimeException(errMsg, throwable);
            }
        }
    }
```